mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -1784,11 +1786,33 @@ def __get_spark_session(self):
         if self.spark_init:
             return self.spark
         try:
+            repo_config = RepoConfig(repo_path=self.repo_path)
+            spark_config = repo_config.get('spark_config')
+
+            from pyspark.conf import SparkConf
             from pyspark.sql import SparkSession
-            self.spark = SparkSession.builder.master(
-                os.getenv('SPARK_MASTER_HOST', 'local')).getOrCreate()
+
+            if spark_config:
+                conf = SparkConf()
+                if repo_config.get('app_name'):
```

Comment:
Did you test this code? It should be `spark_config.get('xxx')`, right?

Reply:
Good catch: made this fix. I encountered some difficulty with connecting to a Yarn cluster to test out this code path so far: (i) from docker to my local Yarn cluster; (ii) setting up a Hadoop/Yarn together with Mage.